os/fs/driver/mtd/smart.c

```diff
@@ -3558,23 +3566,68 @@ static uint16_t smart_findfreephyssector(FAR struct smart_struct_s *dev, uint8_t
 #ifdef CONFIG_SMARTFS_BAD_SECTOR
 			if (dev->badSectorList[x] == FALSE) {
 #endif
-				physicalsector = x;
-				dev->lastallocblock = allocblock;
-				break;
+				ret = MTD_READ(dev->mtd, readaddr, dev->mtdBlksPerSector * dev->geo.blocksize, sector_buff);
```

Comment:
Do not we need to check 'ret' after MTD_READ?

Reply:
Yes, we need to check ret . It is added in https://github.com/Samsung/TizenRT/pull/524/commits/1a45d0e12b357d26bb843cada4b4a856325d9950